cogs/cryochamber.py

```diff
@@ -0,0 +1,72 @@
+import re
+
+from discord import Embed, Colour
+from discord.ext.commands import Cog, CommandError, MessageConverter, TextChannelConverter, group
+
+
+class Cryochamber(Cog):
+    def __init__(self, _):
+        self.ongoing_preservations = set()
+
+    @group(invoke_without_command=True)
+    async def preserve(self, context, *, arg):
+        get_channel = TextChannelConverter().convert
+        get_message = MessageConverter().convert
+        exp = r'(?P<source_specifier>.*?) in (?P<destination_channel_specifier>.*?)'
+        command_match = re.fullmatch(exp, arg)
+
+        if command_match is None:
+            raise CommandError('wrong syntax: ' + arg)
+
+        dest_channel = await get_channel(context,
+                                         command_match.group('destination_channel_specifier'))
+
+        if not dest_channel.permissions_for(context.author).manage_messages:
+            raise MissingPermissions(['manage_messages'])
```

Comment:
This should probably be a Check added to the whole command as a decorator:
https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#checks